[Site Isolation][iOS] Remove hosted layer as subview instead of sublayer

szewai opened the pull request:
#### 93760c4cbf98e14ebf2f588b5f1a1ebe03de4689
<pre>
[Site Isolation][iOS] Remove hosted layer as subview instead of sublayer
<a href="https://bugs.webkit.org/show_bug.cgi?id=269610">https://bugs.webkit.org/show_bug.cgi?id=269610</a>
<a href="https://rdar.apple.com/123113647">rdar://123113647</a>

Reviewed by Alex Christensen.

On iOS, RemoteLayerTreeNode is backed by UIView, so we should modify remote layer tree by updating
the hierachy of UIView. This patch also adds a helper function to avoid this issue in the future.

* Source/WebKit/UIProcess/RemoteLayerTree/RemoteLayerTreeHost.mm:
(WebKit::RemoteLayerTreeHost::updateLayerTree):
(WebKit::RemoteLayerTreeHost::createLayer):
(WebKit::RemoteLayerTreeHost::remotePageProcessCrashed):
* Source/WebKit/UIProcess/RemoteLayerTree/RemoteLayerTreeNode.h:
* Source/WebKit/UIProcess/RemoteLayerTree/RemoteLayerTreeNode.mm:
(WebKit::RemoteLayerTreeNode::addToHostingNode):
(WebKit::RemoteLayerTreeNode::removeFromHostingNode):

Canonical link: <a href="https://commits.webkit.org/275004@main">https://commits.webkit.org/275004@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/6b3a3c4689daee8bf9f179972cd83dd53c3c9d1c

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/40608 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/19621 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/42986 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/43161 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/36697 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/22583 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/16952 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/33690 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/41182 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/16561 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/35006 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/14277 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/14353 "Passed tests") | | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/44435 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/36823 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/36307 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/40048 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/15423 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/12646 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/38380 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/17042 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/17093 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/5399 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/16686 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

achristensen07 commented on Source/WebKit/UIProcess/RemoteLayerTree/RemoteLayerTreeHost.mm:
```diff
@@ -501,7 +492,8 @@ static void recursivelyMapIOSurfaceBackingStore(CALayer *layer)
     for (auto layerID : m_hostedLayersInProcess.take(processIdentifier)) {
-        [layerForID(layerID) removeFromSuperlayer];
+        if (auto* node = nodeForID(layerID))
+            node->setHostingNode(nullptr);
```
I'd call this a different name, like "removeFromHostingNode", then rename setHostingNode to "addToHostingNode" and make it take a RemoteLayerTreeNode&

szewai committed:
[Site Isolation][iOS] Remove hosted layer as subview instead of sublayer
https://bugs.webkit.org/show_bug.cgi?id=269610
rdar://123113647

Reviewed by Alex Christensen.

On iOS, RemoteLayerTreeNode is backed by UIView, so we should modify remote layer tree by updating
the hierachy of UIView. This patch also adds a helper function to avoid this issue in the future.

* Source/WebKit/UIProcess/RemoteLayerTree/RemoteLayerTreeHost.mm:
(WebKit::RemoteLayerTreeHost::updateLayerTree):
(WebKit::RemoteLayerTreeHost::createLayer):
(WebKit::RemoteLayerTreeHost::remotePageProcessCrashed):
* Source/WebKit/UIProcess/RemoteLayerTree/RemoteLayerTreeNode.h:
* Source/WebKit/UIProcess/RemoteLayerTree/RemoteLayerTreeNode.mm:
(WebKit::RemoteLayerTreeNode::addToHostingNode):
(WebKit::RemoteLayerTreeNode::removeFromHostingNode):

Canonical link: https://commits.webkit.org/275004@main
